captcha/storages/base.py

```diff
@@ -0,0 +1,73 @@
+import datetime
+import random
+import time
+import hashlib
+
+from django.utils.encoding import smart_text
+
+from ..models import CaptchaStore
+from ..helpers import get_safe_now
+from ..conf import settings as captcha_settings
+
+
+# Heavily based on session key generation in Django
+# Use the system (hardware-based) random number generator if it exists.
+if hasattr(random, 'SystemRandom'):
+    randrange = random.SystemRandom().randrange
+else:
+    randrange = random.randrange
+MAX_RANDOM_KEY = 18446744073709551616     # 2 << 63
+
+
+class BaseStorage(object):
+    model_class = CaptchaStore
```

Comment:
I think that `model_class` should be set in the DB Storage and not in the base storage. I see that it is used to raise `DoesNotExist` exception but probably will be better if there is a custom exception which will be common for all storages.